File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    n = len(nums)  # 포켓몬 수
    pocketmon = set(nums)
    
    if len(pocketmon) >= n//2:
        return n//2
    
    if n//2 == 1:
        return 1
    
    return len(pocketmon)

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int n = nums.length;
        HashSet<Integer> types = new HashSet<>();
        
        for(int num: nums) {
            types.add(num);
        }
        
        return Math.min(n/2, types.size());
    }
}